File: src/runtime/notify.ts
const MIC_READY_SOUND_PATH = "/usr/share/sounds/freedesktop/stereo/bell.oga";

export const playMicReadyNotification = async (): Promise<void> => {
    const player = Bun.which("pw-play");
    if (!player) return;

    const proc = Bun.spawn([player, MIC_READY_SOUND_PATH], {
        stdout: "ignore",
        stderr: "ignore",
    });
    await proc.exited.catch(() => {}); // optional notification, failure is acceptable
};

File: src/runtime/notify.zig
const std = @import("std");
const cmd = @import("cmd.zig");

const mic_ready_sound_path = "/usr/share/sounds/freedesktop/stereo/bell.oga";

pub fn playMicReadyNotification(allocator: std.mem.Allocator, io: std.Io) void {
    cmd.runDiscard(allocator, io, &.{ "pw-play", mic_ready_sound_path }, 2000) catch {};
}
